From: SeongJae Park <sj@kernel.org>
Date: Wed, 2 Jul 2025 21:22:51 -0700
Subject: [PATCH] ==== damon_initialized() ====

mm/damon: define and use DAMON initialization check function

If DAMON is tried to be used by its API callers when it is not yet
successfully initialized, the callers could be crashed.  Such issues
actually happened and were fixed [1].  DAMON API callers are therefore
having their own hacks for seeing if it is safe to use DAMON or not.
Those built on an untreliable assumption that DAMON should be ready to
be used on module init time.  DAMON initialization could fail if
KMEM_CACHE() fails, though.  Also those are basically duplications that
make their maintenance difficult.

Make it reliable and easy to maintain, by implementing a new DAMON core
layer API function for seeing if DAMON is ready to be used or not, and
replacing the hacks of DAMON API callers with the new core layer
function.

Changes from RFC
(https://lore.kernel.org/20250912023946.62337-1-sj@kernel.org)
- Rebase on latest mm-new

[1] https://lore.kernel.org/20250909022238.2989-1-sj@kernel.org

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 hkml_cv_bogus/hkml_cv_bogus_ayhlq9i2 | 0
 1 file changed, 0 insertions(+), 0 deletions(-)
 create mode 100644 hkml_cv_bogus/hkml_cv_bogus_ayhlq9i2

diff --git a/hkml_cv_bogus/hkml_cv_bogus_ayhlq9i2 b/hkml_cv_bogus/hkml_cv_bogus_ayhlq9i2
new file mode 100644
index 000000000000..e69de29bb2d1
-- 
2.39.5

